docs/applying-terraform.md

```diff
@@ -66,3 +66,57 @@ aws ecs wait services-stable \
  --services "$APPLICATION" \
  --region "$AWS_REGION"
 ```
+
+### Monitoring
+
+The monitoring stack of GOV.UK is in a separate cluster compared to the one where
+the GOV.UK apps run. Below are the details how to run deploy the monitoring stack
+which includes only Grafana for now.
+
+1. Base Infrastructure
+
+```sh
+cd terraform/deployments/monitoring-test/base
+
+gds aws govuk-test-admin -- terraform init
+
+gds aws govuk-test-admin -- terraform apply \
+ -var-file=../../variables/test/common.tfvars \
+ -var-file=../../variables/test/infrastructure.tfvars
+```
+
+2. Grafana Task Definition & Service Update
+
+```sh
+cd terraform/deployments/monitoring-test/grafana
+
+gds aws govuk-test-admin -- terraform init
+
+gds aws govuk-test-admin -- terraform apply
+
+task_definition_arn=$(gds aws govuk-test-admin -- terraform output task_definition_arn)
+
+gds aws govuk-test-admin -- aws ecs update-service  \
+ --cluster monitoring \
+ --service "grafana" \
+ --task-definition "$task_definition_arn" \
+ --region eu-west-1
+
+ aws ecs wait services-stable \
+  --cluster monitoring \
+  --services "grafana" \
+  --region eu-west-1
+```
+
+3. Grafana Internal Configuration
+
+```sh
+cd terraform/deployments/monitoring-test/grafana/app-config
+
+gds aws govuk-test-admin -- terraform init
+
+gds aws govuk-test-admin -- terraform apply
+```
+
+If you get an error: `Error: status: 404, body: {"message":"Data source not found"}`,
+you should run: `gds aws govuk-test-admin -- terraform state rm module.grafana-app-config.grafana_data_source.cloudwatch` before re-applying the terraform.
```

Comment:
Would this error occur frequently? What is the cause of it?

Reply:
This occurs when a new Grafana container is spun and it does not have the Cloudwatch config anymore. It is weird that it didn't just re-create it.

Reply:
Interesting, is that a bug in the grafana provider? 

I wonder if we can prevent it somehow... perhaps with a `depends_on = [grafana_data_source.cloudwatch]`?

Reply:
Yes it is a good point. If for any reason, the container is respun, we lose the config data.

Reply:
That sounds good, sorry it's extra faf - feel free to merge this in and write a card to do that. I'm guessing we could still use the Terraform provider in this case, since it'll just persist whatever changes the resource makes via HTTP calls.